dc/s2n-quic-dc/src/path/secret/map/handshake.rs

```diff
@@ -84,7 +84,16 @@ impl dc::Path for HandshakingPath {
         &mut self,
         session: &impl s2n_quic_core::crypto::tls::TlsSession,
     ) -> Result<Vec<s2n_quic_core::stateless_reset::Token>, s2n_quic_core::transport::Error> {
-        self.application_data = Some(self.map.store.application_data(session));
+        match self.map.store.application_data(session) {
+            Ok(application_data) => {
```

Comment:
I wonder if we should make this Option<Application data> so there's no need for us to have our own dummy we're cloning for the empty case...